test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -59,48 +117,200 @@ defmodule Trento.DeregistrationProcessManagerTest do
+      host_id = UUID.uuid4()
+      sap_system_id = UUID.uuid4()
+      instance_number = "01"
+
+      events = [
+        %DatabaseInstanceRegistered{
+          host_id: host_id,
+          sap_system_id: sap_system_id,
+          instance_number: instance_number
+        }
+      ]
+
+      {commands, state} = reduce_events(events, initial_state)
+
+      assert [] == commands
+
+      assert %DeregistrationProcessManager{
+               database_instances: [
+                 %Instance{
+                   sap_system_id: ^sap_system_id,
+                   instance_number: ^instance_number
+                 }
+               ],
+               application_instances: []
+             } = state
+    end
+
+    test "should add application instance when ApplicationInstanceRegistered event is emitted" do
+      sap_system_id_1 = UUID.uuid4()
+      instance_number_1 = "01"
+
+      initial_state = %DeregistrationProcessManager{
+        database_instances: [],
+        application_instances: [
+          %Instance{
+            sap_system_id: sap_system_id_1,
+            instance_number: instance_number_1
+          }
+        ]
+      }
+
+      host_id = UUID.uuid4()
+      sap_system_id_2 = UUID.uuid4()
+      instance_number_2 = "01"
+
+      events = [
+        %ApplicationInstanceRegistered{
+          host_id: host_id,
+          sap_system_id: sap_system_id_2,
+          instance_number: instance_number_2
+        }
+      ]
+
+      {commands, state} = reduce_events(events, initial_state)
+
+      assert [] == commands
+
+      assert %DeregistrationProcessManager{
+               database_instances: [],
+               application_instances: [
+                 %Instance{
+                   sap_system_id: ^sap_system_id_2,
+                   instance_number: ^instance_number_2
+                 },
+                 %Instance{
+                   sap_system_id: ^sap_system_id_1,
+                   instance_number: ^instance_number_1
+                 }
+               ]
+             } = state
+    end
+
+    test "should update the state with the proper cluster id when ClusterRolledUp event is emitted" do
       initial_state = %DeregistrationProcessManager{}
       cluster_id = UUID.uuid4()
-      host_id = UUID.uuid4()
+      cluster_hosts = [UUID.uuid4(), UUID.uuid4()]
 
-      events = [%HostAddedToCluster{cluster_id: cluster_id, host_id: host_id}]
+      events = [
+        %ClusterRolledUp{cluster_id: cluster_id, snapshot: %Cluster{hosts: cluster_hosts}}
+      ]
 
       {commands, state} = reduce_events(events, initial_state)
 
       assert [] == commands
       assert %DeregistrationProcessManager{cluster_id: ^cluster_id} = state
     end
 
+    test "should update state after when SapSystemRolledUp event received" do
+      sap_system_id = UUID.uuid4()
+      database_instance_number = "00"
+      application_instance_number = "01"
+
+      initial_state = %DeregistrationProcessManager{
+        database_instances: [],
```

Comment:
Maybe add at least one instance to database_instance or application_instance, to see that it adds the values properly